### PR TITLE
tests: add test case for upgrading TiDB

### PR DESCRIPTION
**Pull request by csuzhangxc**
### What problem does this PR solve? 

add test case for upgrading TiDB

### Check List 

Tests 

 - No code

Code changes

Side effects

Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

**Comment by csuzhangxc** [comment]
ha_cass failed: https://internal.pingcap.net/idc-jenkins/blue/organizations/jenkins/dm_ghpr_test/detail/dm_ghpr_test/7562/pipeline

/run-all-tests

File: .github/workflows/upgrade-via-tiup.yml [hunk 1]
```diff
@@ -1,6 +1,10 @@
 name: Upgrade via TiUP
 
 on:
+  pull_request:
+    branches:
+    - master
+    - release-2.0
```
Comment (csuzhangxc) [inline .github/workflows/upgrade-via-tiup.yml:7]: I'll remove these lines after received LGTMs

**Comment by lance6716** [comment]
LGTM

**Review by GMHDBJD — DISMISSED** [review]
LGTM

**Review by lance6716 — APPROVED** [review]

**Comment by lance6716** [comment]
required CI passed, could merge now @csuzhangxc

**Comment by csuzhangxc** [comment]
> required CI passed, could merge now @csuzhangxc

wait for another release-2.0 PR merged first, then no need to _merge from release-2.0_ for this PR's cherry-pick PR. 😂

**Pull request merged**